toString() in original object and toString in proxy are different

danrivcap opened the pull request:
In Proxy object the toString() method returns something different than original object does.

benas commented:
Hi,

Thank you for this PR! Valid point 👍 

I don't see any changes to production code. Is this intended or you have not pushed everything? Currently I see only a failing test added to `RuleProxyTest`.

For the fix, I suggest doing the [same as](https://github.com/j-easy/easy-rules/blob/c2f8c4643b33d631be5883dea322dcc77308b915/easy-rules-core/src/main/java/org/jeasy/rules/core/RuleProxy.java#L133) the `compareTo` method: we should see if there is already a `toString` method. If there is one, we should return it, otherwise use the rule name.

Kr
Mahmoud

benas commented:
I merged the failing test and fixed it. I also added you to the contributors list.

Thank you for your time.

danrivcap commented:
Thanks!!!

I only provided the test, because this weeks I was really busy to fix it.

Sometimes I only have time to hunt the bugs and prove it with a unit test,
and when I have more time I fix them.

Thanks for adding me in contributors list I hope find and hunt more issues
and bugs.

regards
Daniel Rivero Capellán

PS, Sorry for my english

2018-04-08 13:06 GMT+02:00 Mahmoud Ben Hassine <notifications@github.com>:

> I merged the failing test and fixed it. I also added you to the
> contributors list.
>
> Thank you for your time.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/j-easy/easy-rules/pull/144#issuecomment-379541531>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AT9oyi3_SIKGozLp7WScRLE4Y_-X_Uynks5tme8_gaJpZM4TCi3c>
> .
>



-- 
----------------------------------------------------------------
Es duro caer, pero es peor no haber intentado nunca subir.
